<a href="https://colab.research.google.com/github/FitriFatimah6/pemograman-lanjutan-sem3/blob/main/Pendahuluan_Fitri_Fatimah.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#**🖥️👩🏻‍💻PRAKTIKUM PEMROGRAMAN LANJUT🔥**
##**Pertemuan (1) PENDAHULUAN PYTHON - Senin, 14 Oktober 2024**

👤 Nama: Fitri Fatimah / NRP: 2C2230003 / Kelas: III/A Sains Data

💬Deskripsi Pembahasan Materi Pengulangan:

1.1.  Beberapa karakteristik dari bahasa pemrograman Python
- Python bersifat case-sensitive, artinya perbedaan huruf besar dan huruf kecil menyebabkan perbedaan makna sintaks.

- Python sangat memerhatikan indentasi dan pergantian baris. Kesalahan indentasi dapat menyebabkan gagal compile hingga kesalahan program.

- Variabel di python bersifat implisit dan dinamis. Artinya, sebuah variabel tidak perlu dideskripsikan tipe datany

1.2. Macam - macam tipe data :
- Bool(Boolean) : True atau False

- Int(Bilangan bulat) : Seluruh bilangan bulat

- Float(Bilangan real : Seluruh bilangan real

- String(Teks) : Kumpulan karakter




#🤸🏻‍♀️ EKSPLORASI PRIBADI DI KELAS ✨

In [1]:
X = 101
Y= 4

X//Y
X/Y

print(X//Y)
print(X/Y)

25
25.25


In [2]:
# Meminta input dari pengguna
N = int(input("Masukkan nilai N: "))

# Memeriksa apakah N adalah positif, negatif, atau nol
if N > 0:
    if N % 2 == 0:
        print(f"{N} adalah bilangan positif genap")
    else:
        print(f"{N} adalah bilangan positif ganjil")
elif N < 0:
    print(f"{N} adalah bilangan negatif")
else:
    print("N adalah nol")

Masukkan nilai N: 9
9 adalah bilangan positif ganjil


- Interpretasi coding:

- Interpretasi Hasil:
9 adalah bilangan positif ganjil

#🧠 LATIHAN SOAL DI KELAS ⏱️

Problem (1)

- Interpretasi coding:

- Interpretasi Hasil:

Problem (2)

- Interpretasi coding:

- Interpretasi Hasil:

#🏆 POST TEST 🎯

Problem (1)

In [6]:
#H01_2C2230004_01.py
#program asumsi keuntungan tiap barang selalu ada

# Input harga dasar dan harga jual barang
harga_dasar_A = int(input("Masukkan harga dasar barang A: "))
harga_jual_A = int(input("Masukkan harga jual barang A: "))
harga_dasar_B = int(input("Masukkan harga dasar barang B: "))
harga_jual_B = int(input("Masukkan harga jual barang B: "))
harga_dasar_C = int(input("Masukkan harga dasar barang C: "))
harga_jual_C = int(input("Masukkan harga jual barang C: "))

# Hitung keuntungan untuk masing-masing barang
keuntungan_A = harga_jual_A - harga_dasar_A
keuntungan_B = harga_jual_B - harga_dasar_B
keuntungan_C = harga_jual_C - harga_dasar_C

# Tentukan barang dengan keuntungan terbesar
if keuntungan_A > keuntungan_B and keuntungan_A > keuntungan_C:
    print("Barang yang harus ditawarkan adalah barang A")
elif keuntungan_B > keuntungan_A and keuntungan_B > keuntungan_C:
    print("Barang yang harus ditawarkan adalah barang B")
else:
    print("Barang yang harus ditawarkan adalah barang C")


Masukkan harga dasar barang A: 10
Masukkan harga jual barang A: 15
Masukkan harga dasar barang B: 10
Masukkan harga jual barang B: 12
Masukkan harga dasar barang C: 10
Masukkan harga jual barang C: 11
Barang yang harus ditawarkan adalah barang A


- Interpretasi coding:
- Interpretasi hasil:
case 1
- Masukkan harga dasar barang A: 10
- Masukkan harga jual barang A: 15
- Masukkan harga dasar barang B: 10
- Masukkan harga jual barang B: 12
- Masukkan harga dasar barang C: 10
- Masukkan harga jual barang C: 11
- Barang yang harus ditawarkan adalah barang A

case 2
- Masukkan harga dasar barang A: 8
- Masukkan harga jual barang A: 11
- Masukkan harga dasar barang B: 7
- Masukkan harga jual barang B: 12
- Masukkan harga dasar barang C: 6
- Masukkan harga jual barang C: 12
- Barang yang harus ditawarkan adalah barang C

*   Item daftar
*   Item daftar



problem (2)

In [9]:
def kelas_mahasiswa(nrp_akhir):
  nrp_akhir = int(nrp_akhir)

  # menentukan kelas berdasarkan rentang NRP
  if 1 <= nrp_akhir <= 100:
    return "K1" if nrp_akhir % 2 != 0 else "K2"
  elif 101 <= nrp_akhir <= 200:
    return "K3" if nrp_akhir % 2 != 0 else "K4"
  elif 201 <= nrp_akhir <= 300:
    return "K5" if nrp_akhir % 2 != 0 else "K6"
  elif nrp_akhir > 300:
    return "K7" if nrp_akhir % 2 != 0 else "K8"
  else:
    return "NRP Tidak Valid"

# input 3 digit terakhir NRP mahasiswa
nrp = input("Masukkan 3 digit terakhir NRP: ")
kelas = kelas_mahasiswa(nrp)

# menampilkan hasil
print(f"Mahasiswa dengan NRP {nrp} berada di kelas{kelas}.")

Masukkan 3 digit terakhir NRP: 121
Mahasiswa dengan NRP 121 berada di kelasK3.


interpretasi hasil:
case 1
- Masukkan 3 digit terakhir NRP: 121
- Mahasiswa dengan NRP 121 berada di kelas K3.


case 2
- Masukkan 3 digit terakhir NRP: 544
- Mahasiswa dengan NRP 544 berada di kelas K8.

📌Catatan:
  - Notebook python ini dibuat untuk memudahkan temen-temen belajar ngoding dan pengumpulan tugas.
  - Silakan download notebook ini lalu upload di Google Collab milik kalian agar format `pertemuan`, `topik pertemuan`,
`tanggal`, `deskripsi pembahasan materi`, `baris coding` dan `interpretasi` bisa disesuaikan dengan kebutuhan teman-teman.

  - Interpretasi coding dan hasil dimaksudkan agar teman-teman semakin memahami apa yang dipelajari.

  - Deskripsi pembahasan materi dimaksudkan agar teman-teman semakin memahami apa fokus yang dipelajari.

  - Tulisan di setiap sebelum baris code disebut `markdown`. Ada beberapa jenis simbol yang memudahkan kepenulisan `markdown` kita, yaitu seperti : untuk mengatur ukuran tulisan biasanya tulisan pada header menggunakan ini`#`, `##`, `###`, untuk membuat pointer menggunakan ini `-`, untuk membuat tulisan ter-highlight menggunakan ini *` `* , untuk short-cut membuat tulisan jadi **bold** gunakan ini `** **`, dan untuk *tulisan miring* gunakan ini `* *` .

  - Semangat belajar! 🌟✨

©️ LABORATORIUM SAINS DATA 2024